libs/types-rs/src/geometry.rs

```diff
@@ -432,74 +431,6 @@ pub fn find_largest_subset_intersecting_line(
                     .collect::<Vec<_>>(),
             )
         })
-        // Pick the list of rectangles that is the longest.
-        .max_by_key(Vec::len)
```

Comment:
The previous version hardcoded the criterion for picking the best subset to "most number of rectangles". Since we want the criterion to vary depending on which edge we're looking at, I changed this function to return an iterator over all the subsets of rectangles.